ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -56,20 +56,21 @@ export function useTransactionDisplayData (transactionGroup) {
   const knownTokens = useSelector(getTokens)
   const t = useI18nContext()
   const { initialTransaction, primaryTransaction } = transactionGroup
-  // initialTransaction contains the data we need to derive the primary purpose of this transaction group
-  const { transactionCategory } = initialTransaction
+  const transaction = isCancelledTransaction(primaryTransaction) ? initialTransaction : primaryTransaction || initialTransaction
```

Comment:
I think we should probably undo this change and simply override the status variable to use the canceled status when the `primaryTransaction` is canceled?

Reply:
But this recommendation will still result in showing the wrong data that we want since `primaryTransaction.txParams` is different from `initialTransaction.txParams`. And we sort of do this already in https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/helpers/utils/transactions.util.js#L225-L227

Reply:
You're right. I think I am confusing myself sifting through the comments so far and the code. I'm gonna take another look at what you have.